### ISSUE TITLE
Testing UK/DE Alexa

### ISSUE THREAD

**Issue by damien-mclaughlin** [alexa/alexa-avs-sample-app#297]
By default we register with Amazon.com US Alexa and it's a great tutorial, but with the release of the UK and soon DE Alexa can we register with either UK or  DE when it launches to test those versions of Alexa. There are a few related issues, so i didn't want to start fiddling it's not currently possible, so thought i would check here, with the people that might know. Thx

**Comment by tplaskota** [comment]
You can change common settings like  measurment units etc. You also already are able to set address in US, UK or DE. You can pair rpi alexa with android/ios apps anywhere in the world. Only thing I'm not sure is changing to german language. Is there anything in particular you need to know?

Source: I'm from Poland, set up timezone and address to de and using alexa android paired app without problems.

**Comment by damien-mclaughlin** [comment]
Hi tplaskota, thanks for the reply, i live in Berlin, though i am from the UK, so most of my Amazon services atm are tied to .de including Prime, when DE comes online i would like to tie in voice ordering, audible.de, music from Prime, Kindle etc, atm not having a Prime membership with .com leaves these services unsupported in AVS. I also want to test the Deutsche sprache. Not being a native Deutsche speaker and with some services still linked to the UK, the UK Alexa would also prove useful in the interim. But as i understand it from looking at another issue #302 AVS will not support the UK until early 2017.

**Comment by scoobyd00** [comment]
Set this up today and its working but not in the UK due to support not being available until early 2017.
It works for time, and other bits and pieces ok though.

Ive got an issue when using the android app that it keeps on saying that my device is offline??.

Is this due to the fact im in the uk?.

**Comment by damien-mclaughlin** [comment]
You can access the settings for this build from any web browser, if you got to http://alexa.amazon.com/ and then settings you could check if Alexa also shows offline in the web browser so you could verfiy if it is an issue specific to the official UK app or another issue.

**Comment by Croccy22** [comment]
Same problem here.  Have built everything and it's working but it defaulting to Alexa in the USA. So none of my music or skills will work.

Also have a problem where you cannot remove these devices from your Alexa App. If I log onto my amazon.Co.UK account and go to manage devices they do not show up. Spoke to Amazon support about this and they couldn't figure it out either.

Guess it's just another thingreat that doesn't work properly in the UK :(

**Comment by scoobyd00** [comment]
The issue is that avs hasn't been rolled out for the UK yet, so it will default to the US.

Early 2017 is the estimate for it to be working in the UK. Hopefully amazon will get it working before this, fingers crossed.

Looks like the offline issue on the app is a problem as the pi is perm connected to the Alexa servers, it only connects when needed so most of the time it's showing as offline in the app.
Changing the units measures works though, temperatures are in centigrade. Weather location doesn't work, neither does traffic bit these will no doubt work once avs is released for the UK.

**Comment by damien-mclaughlin** [comment]
Interesting that the weather and traffic don't work in the UK, most likely due to the format of UK postcode's not being recognised, none of the ones i put from when i lived in the UK are recognised, but here in Berlin, location, weather and Traffic do work; the postcodes are a similar format to US Zip codes i.e. Berlin 13627

**Comment by Croccy22** [comment]
The weather works fine for me. Not tried traffic yet. Just none of the
skills work as I can only add skills via the alexa app which is linked to
my UK account.

**Comment by Croccy22** [comment]
To enter a UK address in the Alexa app seems broken. You don't appear to be
able to enter a complete post code. But if you type in a street address
then it will find it and auto complete part of the postcode. Some fine user
testing obviously took place :)

**Comment by scoobyd00** [comment]
@Croccy22  Thanks, just tried updating the app with my street name and it looks like its worked.
Ill tedt it tomorrow on the pi to see if it selects the correct weather for my location

**Comment by damien-mclaughlin** [comment]
Yep seems to work, drop's off the last two digits of the postcode, but sets location fine, for Traffic then you would just need to go to settings, traffic, and update the the 'To' field by the same method and Alexa 'should' be able to provide Traffic information

**Comment by scoobyd00** [comment]
Weather is working fine now thanks for @Croccy22 

Has anyone got the 'discover devices' working?.

Im trying to get my hue lights to be controlled?.

**Comment by scoobyd00** [comment]
Just found out that discovery doesnt work on AVS. This is a deal breaker for me.

Looks like ill have to purchase a dot next week.

**Comment by Croccy22** [comment]
Discovery itself does work. But the issue seems to be that this is still
all tied in with the US version of alexa only. If I do discover devices it
finds two devices which are ones I set up before the echo was launched in
the UK. Until we can link this up to our UK accounts it will be pretty
useless :(

**Comment by scoobyd00** [comment]
That's useful to know that in the future I can probably use my pi in another room fully functional.

For the time being I've just ordered a dot to play with in the living room.

**Comment by damien-mclaughlin** [comment]
This is a bit of a side step and not specific to issues with the UK/DE but an issue across the board for the AVSpi you can't use Spotify, Prime Music or the AVSpi as a Bluetooth speaker, it's less of an issue in the US as they have iheartradio whereas the only option outside the US is tunein. But you can add a fairly similar functionality with AirPlay and send music from your phone or computer to your AVSpi Speaker (you'll need an app for Android, i use Allstream): It doesn't use Alexa, but it's a nice functionality to add in the absence of the above services;

Terminal commands below
`sudo apt-get install git libao-dev libssl-dev libcrypt-openssl-rsa-perl libio-socket-inet6-perl libwww-perl avahi-utils libmodule-build-perl`Enter (Installs packages needed)
`git clone https://github.com/njh/perl-net-sdp.git`Enter
`cd perl-net-sdp`Enter
`perl Build.PL`Enter
`sudo ./Build`Enter
`sudo ./Build test`Enter
`sudo ./Build install`Enter
`cd ..`Enter
`git clone https://github.com/hendrikw82/shairport.git` Enter (This installs Shairport)
`cd shairport`Enter
`make`Enter
`./shairport.pl -a AlexaPi`Enter

Credit to Thorin Klosowski

**Comment by damien-mclaughlin** [comment]
Just for info on discovery devices, I tested these TP-Link HS100 smart plugs that i picked up today, both work fine with AVS Alexa once you have added them to your network with the Kasa app and then enabled then to Alexa using the Kasa skill (both of these are new and unregistered previously with either my DE / UK account - so no registration conflicts ) So Alexa has been turning my lights off and on all day :)
https://www.amazon.de/gp/product/B01I59FOIU/ref=oh_aui_detailpage_o00_s00?ie=UTF8&psc=1

**Comment by pfink** [comment]
For all people interested in AVS UK/DE, you could subscribe [here](http://www.amazon.com/gp/html-forms-controller/AVS_Notification_UKDE) to be notified as soon as it's available.

**Comment by umma08** [comment]
I managed to get Alexa to recognise my location, and updated the weather measurements and distance measurements. It now knows my location and settings on reboot. however, it does not recognise any changes i make to the Flash Briefings and does not recognise that my google calendar is linked. Any ideas/fixes? thanks!

**Comment by damien-mclaughlin** [comment]
Hey umma08, if you are based in either the UK / DE and signed up to the Amazon Developer portal with     your usual Amazon ID what is most likely happening is a region conflict where the Alexa app or web portal points to uk / de host but the AVS on the RPI points to the US as there is no official support on the RPI AVS for UK / DE yet. So Alexa will show as offline in the App and web portal and you can't make the changes to flash briefling, skills etc in the Alexa app and have them filter through to the AVS on the RPI as it is pointing to the US Alexa host. For now until there is official UK / DE AVS on RPI the only work around that i use here in Germany is to create a us Amazon.com account and use that to set up a Developer portal account and set then set up AVS on the RPI. You can update anything you need then as when you log into the Alexa app or web portal with you Amazon.com account everything points to the expected host for AVS and changes will filter throught to the RPI. Hope it helps and hopefully UK / DE support for AVS on the RPI is not far away.

**Comment by umma08** [comment]
@damien-mclaughlin yes, i figured this out today - in a slightly different way to you.

First of all i set up a VPN on my normal computer to the United States. Then i cleared all my browser cache and history (to flush any location based cookies for amazon). Then i attempted to log into alexa.amazon.com. This forces me to log in to alexa.amazon.co.uk (most of the time, sometimes i was successful getting into .com - i am not sure why). 

When i was in the home page of the .co.uk settings page it was showing my alexa as offline. I then changed the url to alexa.amazon.com leaving the rest of the url the same and reloaded. 

Low and behold, i got to the settings page in the US site and it was showing my pi as online. 

Here i could change my location, and also add skills. The only problem is that all the skills are from the US site (different from UK skills), and obviously there are no local skills. 

However, i was able to load flash briefings, and the alexa would remember them.

The telegraph flash briefing gives me UK news. 

The BBC and Guardian (which i set through the co.uk setttings page) still appear in my skillset, but do not work - even though they are enabled.  

Long story short, i got the flash briefings and added some skills. 

Curiosity and Magic 8 Ball seem to work well. 

Sadly, RadioPlayer is not available through the US site, and does not enable properly on the UK site. 

Hope that helps some people.

**Comment by damien-mclaughlin** [comment]
@umma08 one of the slightly annoying things for me living in Germany was that prior to the official launch of the echo here in Deutshland, i had everything linked, set up and working connected to my official Amazon account, as soon as it launched officially in DE i could no longer access any of my skills, smart devices etc, and had to go through the above to get it all back.., but hey, it's free and designed for testing so i guess it wouldn't be fair to complain. Not specific to testing UK / DE, but if you have some time spare one of the cooler things i have found to do with this setup involves KODI (in my case on another RPI but should wherever you run Kodi ) It's a bit involved and probably a weekend project, but quiet cool to get Alexa to play Movies, TV, Music etc on your TV (i would recommend the AWS Lamba option with RPI though as the Heroku tends to time out needing a reset of the Java engine ) https://github.com/m0ngr31/kodi-alexa

**Comment by keleric** [comment]
I'm going to close this issue, since it has to do with using AVS in an unsupported way. If you still end up having trouble using AVS in the UK or Germany after those countries officially launch for AVS, please file a new issue! Thanks!
